src/msw/dib.cpp

```diff
@@ -654,12 +748,24 @@ bool wxDIB::Create(const wxImage& image, PixelFormat pf)
         }
         else // no alpha channel
         {
-            for ( int x = 0; x < w; x++ )
+            if ( dstDepth != 1 )
             {
-                *dst++ = src[2];
-                *dst++ = src[1];
-                *dst++ = src[0];
-                src += 3;
+                for ( int x = 0; x < w; x++ )
+                {
+                    *dst++ = src[2];
+                    *dst++ = src[1];
+                    *dst++ = src[0];
+                    src += 3;
+                }
+            }
+            else
+            {
+                for ( int x = 0; x < w; x++ )
+                {
+                    wxASSERT(src[0] == 0 || src[0] == 1);
```

Comment:
I think it's a bad idea to have these checks inside the loop, `wxASSERT` is present in production builds too (it's just inactive there by default) and it just doesn't seem right to have it inside an inner loop.

Reply:
done